---
 Exercise 1: Find the remainder

1) ---

 $1944^{2024} \mod 7$

step 1 reduce $1944 \mod 7$

In [52]:
1944/7

277.7142857142857

In [53]:
(277.7142857142857 - 277) *7

5.000000000000057

In [54]:
277 * 7 + 5

1944

$1944 \equiv 5 \mod 7$

now we can use Fermat's Little Theorem 

to clacualte $5^{2024}\mod 7$

we get that $5^{6}\equiv 1 \mod 7$

we can then Reduce the exponent

so the remader is 2 thus $2024\equiv 2 \mod 6$


so we have $5^2 \equiv 5^{2024} \equiv 1944^{2024} \mod 7$

<h3>answer</h3> 

$25 \mod 7 = 4$

2) ---

$6^{1001}+1 \mod 7$

frist lets find

$6^{1001}\mod 7 \equiv x$

with fermats littl therom we get <br>
$6^6 \equiv 1 \mod 7$

$1001 \mod 6$ = 5

$6^{5}+1 \mod 7$

<h3>answer</h3>

$7777 \mod 7$ = 0 

3) ---

$1532^{5}-1 \mod 9$

In [55]:
1532 % 9

2

$2^{5}-1 \mod 9$

In [56]:
2**5 -1

31

$31 \mod 9$

In [57]:
31 % 9

4

<h3>answer</h3>

$31 \mod 9 \equiv 4$

4) ---

$3^{2024} \mod 13$

with fermats littl therom we get <br>
$3^{12} \equiv 1 \mod 13$

In [58]:
2024 % 12

8

$3^{8} \mod 13$

In [59]:
3**8

6561

$6561 \mod 13$

In [60]:
6561 % 13

9

<h3>answer</h3>

$6561 \mod 13 \equiv 9$

5) ---

$2012^{2013} + 2014^{2015} \mod 7$

In [61]:
2012 % 7

3

In [62]:
2014 % 7

5

$3^{2013} + 5^{2015} \mod 7$

with fermats littl therom we get <br>
$3^{6} \equiv 1 \mod 7$ and $5^{6} \equiv 1 \mod 7$

In [63]:
2013 % 6 

3

In [64]:
2015 % 6 

5

$3^{3} + 5^{5} \mod 7$

In [65]:
3**3 + 5**5

3152

$3152 \mod 7$

In [66]:
3152 %7

2

<h3>answer</h3>

$2012^{2013} + 2014^{2015} \mod 7  \equiv 2$

6) --- 

$2014^{2015}+2015^{2016} \mod 17$

In [67]:
2014 % 17

8

In [68]:
2015 % 17

9

with fermats littl therom we get <br>
$8^{16} \equiv 1 \mod 17$ and $9^{16} \equiv 1 \mod 17$

In [69]:
2015 %16

15

In [70]:
2016 %16

0

$2014^{2015}+2015^{2016} \mod 17 \equiv 8^{15}+1 \mod 17$

In [71]:

(8**15 + 1) % 17

16

<h3>answer</h3>

$2014^{2015}+2015^{2016} \mod 17 \equiv 16$

Exercise 2: prove that 
---



---
$2222^{5555} + 5555^{22222} \mod7 = 0 $

In [72]:
2222 % 7

3

In [73]:
5555 % 7

4

$3^{5555} + 4^{22222} \mod7 $

with fermats littl therom we get <br>
$3^{6} \equiv 1 \mod 7$ and $4^{6} \equiv 1 \mod 7$

In [74]:
5555 % 6

5

In [75]:
22222 % 6

4

$3^{5} + 4^{4} \equiv 2222^{5555} + 5555^{22222} \mod7 $

In [76]:
3**5 + 4**4

499

$499 \mod7 $

In [77]:
499 % 7

2

this is not true i suspect this is a typo in the slides<br>
this equation might be this inseted $2222^{5555} + 5555^{2222} \mod7 = 0 $ <br>
in such case it would be  $3^{5} + 4^{2} \equiv 2222^{5555} + 5555^{2222} \mod7 $

In [78]:
3**5 + 4**2

259

In [79]:
259 / 7

37.0

$2222^{5555} + 5555^{2222} \mod7 = 0 $ <br> 
is true <br> 
$2222^{5555} + 5555^{22222} \mod7 = 0 $ <br> is not 

---
$5^{70}+7^{50} \mod 12 = 2$

In [80]:
70 % 11

4

In [81]:
50 % 11

6

---
$776^{776} + 777^{777} + 778^{778} \mod 5 = 2$

we whant to show that $776^{776} + 777^{777} + 778^{778} \equiv 2 \mod 5$

$5 \vert (776^{776} + 777^{777} + 778^{778} - 2)$

$5 \vert (1^{776} + 2^{777} + 3^{778} - 2)$


we can use fermats little therom to reduse the exponents

$5 \vert (1^{0} + 2^{1} + 3^{2} - 2)$

In [100]:
(1**0)+(2**1)+(3**2)-2

10

$5 \vert (10)$

this is true

---
$3^{2024} + 4^{2024} \mod 13 = 12$

$3^{2024} + 4^{2024} \mod 13$

In [96]:
2024%12

8

$3^{8} + 4^{8} \mod 13$

$3^{8} + 4^{8} \equiv 12 \mod 13$

$13 \vert (3^{8} + 4^{8} - 12) $

In [97]:
3**8 + 4**8 - 12

72085

In [98]:
72085 / 13

5545.0

this is true 

---
$2013^{2014}+2014^{2015} \mod 4 = 1$

In [94]:
2013 % 4

1

In [95]:
2014 % 4

2

$1^{2014}+2^{2015} \equiv 2013^{2014}+2014^{2015} \mod 4$

$1+2^{2015} \mod 4$

fermats little

$2^{3}\equiv 1 \mod 4$

In [102]:
2015%3

2

$1+2^{2} \mod 4$

Exercise 3: let n be an Integer. prove that
---


a
---

$n^2 \equiv 0$ or $n^2 \equiv 1 (\mod 4) $

we have sevral cases that could be posable 

if n is even $n = 2k$<br>
if n is odd $n = 2k+1$

$(2k)^2 = 4k^2 \equiv 0 \mod 4$<br>
$(2k+1)^2 =(2k+1)(2k+1) = 4k^2+2k+2k+1 = 4k^2+4k+1\equiv 1 \mod 4$<br>

 $4k^2+4k+1\equiv 1 \mod 4$<br>
 $4k^2 \equiv 0 \mod 4$<br>
 $4k \equiv 0 \mod 4$<br>
 Therefor<br>
  $0+0+1\equiv 1 \mod 4$<br>

b
---

$n^3 \equiv 0$ or $n^3 \equiv 6 $ or $n^3 \equiv 1 (\mod 7)$



if: $n \equiv 0 \pmod{7}$

$n^3 \equiv 0^3 \equiv 0 \pmod{7}$



if: $n \equiv \pm 1 \pmod{7}$

$n^3 \equiv (\pm 1)^3 \equiv \pm 1 \pmod{7}$

Since $-1 \equiv 6 \pmod{7}$ and $1 \equiv 1 \pmod{7}$, we have $n^3 \equiv 1 \text{ or } 6 \pmod{7}$.



if: $n \equiv \pm 2 \pmod{7}$

$n^3 \equiv (\pm 2)^3 \equiv \pm 8 \pmod{7}$

Since $8 \equiv 1 \pmod{7}$ and $-8 \equiv -1 \equiv 6 \pmod{7}$, we have $n^3 \equiv 1 \text{ or } 6 \pmod{7}$.



if : $n \equiv \pm 3 \pmod{7}$

$n^3 \equiv (\pm 3)^3 \equiv \pm 27 \pmod{7}$

Since $27 \equiv 6 \pmod{7}$ and $-27 \equiv -6 \equiv 1 \pmod{7}$, we have $n^3 \equiv 1 \text{ or } 6 \pmod{7}$.



Exercise 4:
---


a
---

Prove that ac ≡ bc (mod m) and gcd(c,m) = 1, then a ≡ b (mod m).

let $k \epsilon Z$

we can say that<br>
$ac-bc=km$<br>
becaus gcd(c,m) = 1 <br>


 $ac-bc | c $ -> $km|c$

k = dc<br>
$(a-b) = dm -> a ≡ b (\mod m)$


Exercise 5:
---



Let p, p + k, p + 2k be prime numbers. Prove that: k is multiple of 6.

we need to show that:
$p \equiv 0 \mod 6$

what we know:<br>
$k \epsilon Z$<br>


$GCD(k,p) = p , 1$

this means that the $LCM(k,p) = 1$

we know that p - (p+k)= k where k is and even number becaus the sum of 2 odd numbers is odd

sence the sum of $(p + k) + (p + 2k) = 2p + 3k$ is also and even number

k is an even number $k=2n$

3k is also even so we can express it as a diffrent number 3k = 2m

we now can find the answer by sloving for $3K = 3(2n) = 2m$

$3K = 6n$ where n is some z


Exercise 7:
---

Is there exist the prime number p such that p + 2, p + 4, p + 8, p + 16
are prime numbers ?

p = 3

exp <br>
3+2 = 5 <br>
3+4 = 7 <br>
3+8 = 11 <br>
3+16 = 19 <br>


Exercise 8:
---

Find all the integer number pairs (x, y) such that:
$5^x + 1 = 2^y$

$1 = 2^y-5^x$

x=0 y=1

 if we take mod 4

$5^x + 1 = 2^y(\mod 4)$

$1 ≡ 5^x (\mod 4)$

$1+1 = 2^y (\mod 4)$

which means that y must be 1 so if we solve for x we get that

$5^x + 1 = 2$

$5^x  = 1$

$x=0$

so (x=0 y=1)


Exercise 9:
---

Solve the following linear congruences <br>


$8x \equiv 6 (mod 14)$<br>
---

In [82]:
math.gcd(8,14)

NameError: name 'math' is not defined

In [ ]:
6%2 ==0

True

$4x \equiv 3 (mod 7)$

In [ ]:
for i in range(0,10):
    print(3-(i*7))

3
-4
-11
-18
-25
-32
-39
-46
-53
-60


$4x \equiv -4 (mod 7)$

$x \equiv -1 (mod 7)$

$x=7t-1$ or $x=7t+6$


$x \equiv 6 (mod 7)$

$66x \equiv 100 (mod 121)$<br>
---

In [ ]:
math.gcd(66,121)

11

In [ ]:
100%11 ==0

False

this has no solution

$21x \equiv 14 (mod 91)$<br>
---

In [ ]:
math.gcd(21,91)

7

In [ ]:
math.gcd(21,14,91)

7

$3x \equiv 2 (mod 13)$<br>

$3x \equiv 15 (mod 13)$<br>

$x \equiv 5 (mod 13)$<br>

$x = 13k + 5$

Exercise 10:
---

a
---

$x \equiv 1 (mod 4)$

$x \equiv 2 (mod 3)$

$x \equiv 3 (mod 5)$

$x= k \mod 4 (3*5) + n \mod 3 (4*5)+ m \mod 5 (4*3)$

k = 3,n=1 ,m=4

In [ ]:
r = 15*3+20+12*4
r

113

In [ ]:
check = r % 4 == 1
print(check)
check = r % 3 == 2
print(check)
check = r % 5 == 3
print(check)


True
True
True


x=113

b
---

$x \equiv 2 (mod 3)$

$x \equiv 4 (mod 5)$

$x \equiv 6 (mod 13)$

we can re right this ad an equations frist

$x \equiv 2 (mod 3)$ -> $x = 3n+2$ 

$x \equiv 4 (mod 5)$ -> $x= 5n+4$

$x \equiv 6 (mod 13)$ -> $x = 13n+6$

In [ ]:
from math import lcm
lcm(3,5,13)

195

In [ ]:
from math import gcd
gcd(3,5,13)

1

In [ ]:
def Minvers(m_, M_):
    i = 1 
    while((M_ * i)% m_ != 1):
        i+=1
    return i
    


a1 = 2
a2 = 4
a3 = 6

m1 = 3
m2 = 5
m3 = 13

M = m1*m2*m3
M1 = M/m1
M2 = M/m2
M3 = M/m3


## this is the chinese remander therom
x = a1*M1*Minvers(m1,M1)+a2*M2*Minvers(m2,M2)+a3*M3*Minvers(m3,M3)
x

1514.0

In [ ]:
#checking the results
1514.0 % 3 == 2

True

In [ ]:
1514.0 % 5 == 4

True

In [ ]:
1514.0 % 13 == 6

True

answer is x = 1514

Exercise 11:
---


a : $2^2 * 3^3 * 5^5, 2^5 * 3^3 * 5^2$
---


GCD = $2^2*3^3*5^2$<br>
lcm = 2700000.0

In [ ]:
2**2*3**3*5**2

2700

In [ ]:
(2**2 * 3**3 * 5**5* 2**5 * 3**3 * 5**2) / (2**2*3**3*5**2)

2700000.0

b: $2*3*5*7*11*13, 2^{11} * 3^{9} * 11 * 17^14$
---


$2*3*5*7*11*13,$<br> 
$2^{11} * 3^{9} * 11 * 17^{14}$

gcd = $2 * 3 * 11$<br>
LCM = $(2*3*5*7*11*13)(2^{11} * 3^{9} * 11 * 17^{14}) / gcd$

In [ ]:
gc = 2 * 3 * 11
gc

66

In [ ]:
lc = (2*3*5*7*11*13)*(2**11 * 3**9 * 11 * 17**14) / gcd
lc

3.3971148195111497e+28

c $17,17^{17}$
---


gcd = 17
lmc = 17^{18} / 17

d $2^2 *7,5^3*13$
---


gcd = 1
lcm = 2^2 *7*5^3*13/1 = 45500.0

e 0,5
--

gcd = 5 
Lcm = 0

f $2*3*5*7,2*3*5*7$
--

gcd = 210
lcm = 2*3*5*7*2*3*5*7 / 210 = 210

the code portion is in the c file i have a comand line interface that you can use to test it 

i will past the code in a cell for you to see

In [ ]:

#include <stdio.h>



int GCD(int a , int b){
    while(b != 0){
        int t=b;
        b = a%b;
        a=t;
    }
    return a;    
}

int LCM(int a, int b){
    return (a*b / GCD(a,b));
}

int main(){

    int option;
    printf("Enter a number:[1 for GCD] or [2 for LCM]\n");
    scanf("%d",&option);


    int num1;
    int num2;

    printf("Enter a 2 numbers\n");
    scanf("%d",&num1);
    scanf("%d",&num2);

    if(option == 1){
        printf("GCD = %d \n",GCD(num1,num2));
    }else if(option == 2){
        printf("LCM = %d \n",LCM(num1,num2));
    }else{
        printf("%d is not a valid option",option);
    }

}